connect/runtime/src/test/java/org/apache/kafka/connect/integration/ExampleConnectIntegrationTest.java

```diff
@@ -114,8 +116,7 @@ public void testProduceConsumeConnector() throws Exception {
         // start a sink connector
         connect.configureConnector(CONNECTOR_NAME, props);
 
-        waitForCondition(() -> connect.connectorStatus(CONNECTOR_NAME).tasks().size() == 3
-                        && connectorHandle.tasks().stream().allMatch(th -> th.partitionsAssigned() == 1),
+        waitForCondition(this::partitionsAssigned,
```

Comment:
This appears in the two connector integration tests in AK's codebase. Should `TestUtils` or `EmbeddedConnectCluster` have a method that does this?

Reply:
Would it also be helpful here to catch any exceptions that might be thrown by the condition?

Reply:
`TestUtils` is in the clients module. And adding it to `EmbeddedConnectCluster` seemed incorrect as this method depends on checking that depended on `ConnectorHandle`. I've moved it to an IntegrationTestUtils class. Let me know what you think.